spark/src/main/scala/io/delta/sharing/spark/DeltaSharingSource.scala

```diff
@@ -249,27 +348,27 @@ case class DeltaSharingSource(
    * @param endOffset - Offset that signifies the end of the stream.
    * @return the created DataFrame.
    */
-  private def getFileChangesAndCreateDataFrame(
-    startVersion: Long,
-    startIndex: Long,
-    isStartingVersion: Boolean,
-    endOffset: DeltaSharingSourceOffset): DataFrame = {
-    if (options.readChangeFeed) {
-      throw DeltaSharingErrors.CDFNotSupportedInStreaming
-    }
-
-    maybeGetFileChanges(startVersion, startIndex, isStartingVersion)
-
+  private def createDataFrameFromOffset(
+      startVersion: Long,
+      startIndex: Long,
+      isStartingVersion: Boolean,
+      endOffset: DeltaSharingSourceOffset): DataFrame = {
     val fileActions = sortedFetchedFiles.takeWhile {
-      case IndexedFile(version, index, _, _) =>
+      case IndexedFile(version, index, _, _, _, _) =>
         version < endOffset.tableVersion ||
           (version == endOffset.tableVersion && index <= endOffset.index)
     }
     sortedFetchedFiles = sortedFetchedFiles.drop(fileActions.size)
     // Proceed the offset as the files before the endOffset are processed.
     previousOffset = endOffset
 
-    createDataFrame(fileActions)
+    val filteredActions = fileActions.filter{ indexedFile => indexedFile.getFileAction != null }
```

Comment:
consider adding a comment on when the indexedFile may have a null action.

Reply:
added.